server/sonar-server/src/main/java/org/sonar/server/user/ws/CurrentAction.java

```diff
@@ -61,87 +67,65 @@ public void define(NewController context) {
 
   @Override
   public void handle(Request request, Response response) throws Exception {
-    try (DbSession dbSession = dbClient.openSession(false)) {
-      Optional<UserDto> user = Optional.empty();
-      Collection<String> groups = emptyList();
-      if (userSession.isLoggedIn()) {
+    UserDto user;
+    boolean showOnboarding;
+    Collection<String> groups;
+    if (userSession.isLoggedIn()) {
+      try (DbSession dbSession = dbClient.openSession(false)) {
         user = selectCurrentUser(dbSession);
+        showOnboarding = showOnboarding(user, dbSession);
         groups = selectGroups(dbSession);
       }
-      writeResponse(response, user, groups);
+    } else {
+      user = null;
+      showOnboarding = false;
+      groups = emptyList();
     }
+    writeProtobuf(toWsResponse(user, showOnboarding, groups), request, response);
   }
 
-  private Optional<UserDto> selectCurrentUser(DbSession dbSession) {
-    return Optional.ofNullable(dbClient.userDao().selectActiveUserByLogin(dbSession, userSession.getLogin()));
+  private boolean showOnboarding(UserDto user, DbSession dbSession) {
+    if (settings.getBoolean(CoreProperties.SKIP_ONBOARDING_TUTORIAL)) {
+      return false;
+    }
+    if (dbClient.userDao().selectOnboarded(dbSession, user)) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1126)